sentry-logback/src/main/java/io/sentry/logback/SentryAppender.java

```diff
@@ -0,0 +1,190 @@
+import io.sentry.core.SentryLevel;
+import io.sentry.core.SentryOptions;
+import io.sentry.core.protocol.Message;
+import io.sentry.core.protocol.SdkVersion;
+import java.util.*;
+import java.util.stream.Collectors;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+/** Appender for logback in charge of sending the logged events to a Sentry server. */
+public final class SentryAppender extends UnsynchronizedAppenderBase<ILoggingEvent> {
+  private @Nullable String dsn;
+  private @Nullable String environment;
+  private @Nullable Integer maxBreadcrumbs;
+  private @Nullable Integer shutdownTimeoutMillis;
+  private @Nullable Integer flushTimeoutMillis;
+  private @Nullable Integer readTimeoutMillis;
+  private @Nullable Double sampleRate;
+  private @Nullable Boolean bypassSecurity;
+  private @Nullable Boolean debug;
+  private @Nullable Boolean attachThreads;
+  private @Nullable Boolean attachStacktrace;
+
+  private @NotNull final HubAdapter hubAdapter;
+
+  SentryAppender(@NotNull HubAdapter hubAdapter) {
+    this.hubAdapter = hubAdapter;
+  }
+
+  public SentryAppender() {
+    this(HubAdapter.getInstance());
+  }
+
+  @Override
+  public void start() {
+    if (dsn != null) {
+      Sentry.init(
+          options -> {
+            options.setDsn(dsn);
+            Optional.ofNullable(maxBreadcrumbs).ifPresent(options::setMaxBreadcrumbs);
+            Optional.ofNullable(environment).ifPresent(options::setEnvironment);
+            Optional.ofNullable(shutdownTimeoutMillis).ifPresent(options::setShutdownTimeout);
+            Optional.ofNullable(flushTimeoutMillis).ifPresent(options::setFlushTimeoutMillis);
+            Optional.ofNullable(readTimeoutMillis).ifPresent(options::setReadTimeoutMillis);
+            Optional.ofNullable(sampleRate).ifPresent(options::setSampleRate);
+            Optional.ofNullable(bypassSecurity).ifPresent(options::setBypassSecurity);
+            Optional.ofNullable(debug).ifPresent(options::setDebug);
+            Optional.ofNullable(attachThreads).ifPresent(options::setAttachThreads);
+            Optional.ofNullable(attachStacktrace).ifPresent(options::setAttachStacktrace);
+            options.setSentryClientName(BuildConfig.SENTRY_LOGBACK_SDK_NAME);
+            options.setSdkVersion(createSdkVersion(options));
+          });
+    }
+    super.start();
+  }
+
+  @Override
+  protected void append(@NotNull ILoggingEvent eventObject) {
+    hubAdapter.captureEvent(createEvent(eventObject));
+  }
+
+  /**
+   * Creates {@link SentryEvent} from Logback's {@link ILoggingEvent}.
+   *
+   * @param loggingEvent the logback event
+   * @return the sentry event
+   */
+  @SuppressWarnings("JdkObsolete")
+  final @NotNull SentryEvent createEvent(@NotNull ILoggingEvent loggingEvent) {
+    final SentryEvent event = new SentryEvent(new Date(loggingEvent.getTimeStamp()));
+    final Message message = new Message();
+    message.setFormatted(loggingEvent.getFormattedMessage());
```

Comment:
Do we also have the template? That's what wil be used when grouping if we provide it:

```suggestion
    message.setMessage(loggingEvent.getTemplate());
```

The formatted message is used to display to the user since the actual params are already interpolated

Reply:
That's cool, I didn't know how grouping works.